res/controllers/Reloop-Ready-scripts.js

```diff
@@ -475,485 +478,467 @@ ReloopReady.AutoLoopPadMode = function(index) {
+        this.pads[5] = new components.Button({
+            key: "loop_double",
+            on: ReloopReady.padColorPalette.Green.lit,
+            off: ReloopReady.padColorPalette.Green.dim,
+        });
+        this.pads[6] = new components.Button({
+            inKey: "loop_move",
+            on: ReloopReady.padColorPalette.Cyan.lit,
+            off: ReloopReady.padColorPalette.Cyan.dim,
+            unshift: function() {
+                this.input = ReloopReady.makeButtonDownInputHandler(function() {
+                    this.inSetValue(-loopMoveBeats);
+                });
+            },
+            shift: function() {
+                this.input = ReloopReady.makeButtonDownInputHandler(function() {
+                    this.inSetValue(-engine.getValue(this.group, "loop_size"));
+                });
+            },
+            trigger: ReloopReady.makeIsPressedTrigger(),
+        });
+        this.pads[7] = new components.Button({
+            inKey: "loop_move",
+            on: ReloopReady.padColorPalette.Cyan.lit,
+            off: ReloopReady.padColorPalette.Cyan.dim,
+            unshift: function() {
+                this.input = ReloopReady.makeButtonDownInputHandler(function() {
+                    this.inSetValue(loopMoveBeats);
+                });
+            },
+            shift: function() {
+                this.input = ReloopReady.makeButtonDownInputHandler(function() {
+                    this.inSetValue(engine.getValue(this.group, "loop_size"));
+                });
+            },
+            trigger: ReloopReady.makeIsPressedTrigger(),
+        });
+
+        this.pads.forEach((pad, i) => pad.midi = [0x94 + deckIdx, 0x14 + i]);
 
-    // var baseOffset = index * 8;
+        this.parameterLeft = new components.Button({
+            midi: [0x94 + deckIdx, 0x28],
+            shiftOffset: 0x2,
+            input: ReloopReady.makeButtonDownInputHandler(function() {
+                loopMoveBeats *= 0.5;
+            }),
+        });
+        this.parameterRight = new components.Button({
+            midi: [0x94 + deckIdx, 0x29],
+            shiftOffset: 0x2,
+            input: ReloopReady.makeButtonDownInputHandler(function() {
+                loopMoveBeats *= 2;
+            }),
+        });
 
-    this.pads = this.pads.map((_, i) =>
-        new components.SamplerButton({
-            midi: [0x94 + index, 0x14 + i],
-            // number: baseOffset + i + 1,
+        this.reconnectComponents(c => {
+            if (c.group === undefined) {
+                c.group = group;
+            }
+        });
+    }
+};
+ReloopReady.SamplerPadMode = class extends ReloopReady.PadMode {
+    constructor(deckIdx) {
+        super(deckIdx);
+        this.pads = this.pads.map((control, i) => new components.SamplerButton({
+            midi: [0x94 + deckIdx, control],
             number: i + 1,
             shiftOffset: 0x8,
             off: 0x00,
             outConnect: false,
-        })
-    );
+        }));
+    }
```

Comment:
I noticed that in *Sampler mode*, when a sample (i.e., track) is selected by pressing a pad, irrespective whether this pad is on the left deck or on the right deck, the sample will get assigned to the same numerical pad on *both* decks.

I don't know if this is the UX behavior as intended by [`components.SamplerButton`](https://github.com/mixxxdj/mixxx/wiki/Components-JS#samplerbutton).

But maybe for the Reloop Ready with its 2 × 8 pads = 16 pads, it would make sense to map the sampler buttons 1 through 8 to the *left* deck pads, and map the sampler buttons 9 through 16 to the *right* deck pads (the *Deere* and *LightNight* skins allow to view 16 samplers through their *skin settings*).

Reply:
I think its a matter of which deck the track is currently playing on. However, I do want to map the parameter buttons to cycle through the different available samplers. would that work for you?